src/covariance.h

```diff
@@ -157,15 +170,40 @@ matrix<T> get_covariance_lower_chol(const vector<T>& theta, int n_visits, std::s
   return result;
 }
 
+// Creates a new spatial covariance cholesky.
+template <class T>
+matrix<T> get_spatial_covariance_lower_chol(const vector<T>& theta, const matrix<T>& distance, std::string cov_type) {
+  matrix<T> result;
+  if (cov_type == "gp_exp") {
+    result = get_spatial_exponential<T>(theta, distance);
+  } else {
+    Rf_error(("Unknown covariance type '" + cov_type + "'.").c_str());
+  }
+  return result;
+}
+
 // Creates a grouped correlation object dynamically.
 template <class T>
-matrix<T> get_cov_lower_chol_grouped(const vector<T>& theta, int n_visits, std::string cov_type, int n_groups) {
+matrix<T> get_cov_lower_chol_grouped(const vector<T>& theta, int n_visits, std::string cov_type, int n_groups, int spatial) {
```

Comment:
```suggestion
matrix<T> get_cov_lower_chol_grouped(const vector<T>& theta, int n_visits, std::string cov_type, int n_groups, bool spatial) {
```
should work? would be a bit nicer than int

Reply:
also better consistently use `is_spatial` as argument and variable name